### ISSUE TITLE
Serializaiton crash with amqp backend

### ISSUE THREAD

**Issue by eviltnan** [celery/celery#3121]
Get serialization error while trying to get the async result with amqp backend:

```
In [4]: r.status
---------------------------------------------------------------------------
error                                     Traceback (most recent call last)
<ipython-input-4-8614be5289d4> in <module>()
----> 1 r.status

.virtualenvs/damus/local/lib/python2.7/site-packages/celery/result.pyc in state(self)
    392 
    393         """
--> 394         return self._get_task_meta()['status']
    395     status = state
    396 

.virtualenvs/damus/local/lib/python2.7/site-packages/celery/result.pyc in _get_task_meta(self)
    337     def _get_task_meta(self):
    338         if self._cache is None:
--> 339             return self._maybe_set_cache(self.backend.get_task_meta(self.id))
    340         return self._cache
    341 

.virtualenvs/damus/local/lib/python2.7/site-packages/celery/backends/amqp.pyc in get_task_meta(self, task_id, backlog_limit)
    161         with self.app.pool.acquire_channel(block=True) as (_, channel):
    162             binding = self._create_binding(task_id)(channel)
--> 163             binding.declare()
    164 
    165             prev = latest = acc = None

.virtualenvs/damus/local/lib/python2.7/site-packages/kombu/entity.pyc in declare(self, nowait)
    520         if self.exchange:
    521             self.exchange.declare(nowait)
--> 522         self.queue_declare(nowait, passive=False)
    523 
    524         if self.exchange and self.exchange.name:

.virtualenvs/damus/local/lib/python2.7/site-packages/kombu/entity.pyc in queue_declare(self, nowait, passive)
    546                                          auto_delete=self.auto_delete,
    547                                          arguments=self.queue_arguments,
--> 548                                          nowait=nowait)
    549         if not self.name:
    550             self.name = ret[0]

.virtualenvs/damus/local/lib/python2.7/site-packages/amqp/channel.pyc in queue_declare(self, queue, passive, durable, exclusive, auto_delete, nowait, arguments)
   1252         args.write_bit(auto_delete)
   1253         args.write_bit(nowait)
-> 1254         args.write_table(arguments)
   1255         self._send_method((50, 10), args)
   1256 

.virtualenvs/damus/local/lib/python2.7/site-packages/amqp/serialization.pyc in write_table(self, d)
    348         for k, v in items(d):
    349             table_data.write_shortstr(k)
--> 350             table_data.write_item(v, k)
    351         table_data = table_data.getvalue()
    352         self.write_long(len(table_data))

.virtualenvs/damus/local/lib/python2.7/site-packages/amqp/serialization.pyc in write_item(self, v, k)
    364             self.write(pack('>cd', b'd', v))
    365         elif isinstance(v, int_types):
--> 366             self.write(pack('>ci', b'I', v))
    367         elif isinstance(v, Decimal):
    368             self.write(b'D')

error: 'i' format requires -2147483648 <= number <= 2147483647

```

**Comment by ask** [comment]
What version is this?

**Comment by ask** [comment]
Fixed by py-amqp commit: celery/py-amqp@67209e50816fff46fdae8ce7c990d12da6eb90f1
